src/nvim/if_cscope.c

```diff
@@ -1772,11 +1777,14 @@ static int cs_read_prompt(int i)
       if (bufpos < maxlen - 1 && vim_isprintc(ch)) {
         // lazy buffer allocation
         if (buf == NULL) {
-          buf = xmalloc(maxlen);
+          assert(maxlen >= 0);
+          buf = xmalloc((size_t)maxlen);
         }
         {
           /* append character to the message */
-          buf[bufpos++] = ch;
+          assert(ch >= 0);
+          assert(ch <= CHAR_MAX);
```

Comment:
It is planned to replace those range-checking conditions by dedicated macros at some point.
Then, it would read like this: `assert(IN_RANGE(INT, ch));`, which wouldn't need further explanation.
For asserts requiring explanation, comments are preferred.